python/dgl/distributed/graph_services.py

```diff
@@ -133,7 +133,7 @@ def merge_graphs(res_list, num_nodes):
         dst_tensor = res_list[0].global_dst
         eid_tensor = res_list[0].global_eids
     g = graph((src_tensor, dst_tensor),
-              restrict_format='coo', num_nodes=num_nodes)
+              formats=('coo',), num_nodes=num_nodes)
```

Comment:
do we need to specify formats here? @jermainewang the graph here is a subgraph generated from sampling.

Reply:
You can compare the speed of restricting `coo` and not restricting format after minjie finished the refactor the scheduler.

Reply:
btw `formats` also accepts a single string as input: `formats='coo'`

Reply:
does it work?

Reply:
i see. it doesn't work in `shared_memory`.

Reply:
Why it doesn't work with shared_memory?